lib/TestReportHandler.rb

```diff
@@ -127,11 +127,23 @@ def pass
     return 
   end 
 
+#	Might need to alter ‘score’ calculation based on including/not including setup or teardown failures
+# Or, add the option to decide whether to include them
+
   def finalize
     report.status = 'completed'
     report.score = ((1 - @num_failures / @num_tests.to_f) * 100).round(2)
     report.result = (report.score == 100 ? 'pass' : 'fail')
     report.issued = DateTime.now
+    
+    num_passes = 0
+    report.test[0].action.each do |ac|
```

Comment:
```suggestion
    report.test.each do |test| 
          test.action.each do |ac| 
               [your code here]
          end 
    end 
```
Even though a lot of the TestScripts we are using only have one 'Test' element, there could realistically be any number of them -- so you want to iterate through instead of just inspecting the first one.

Reply:
Additionally, this gives you some safer iteration. We'd fail out on a '`NilClass` does not have method `action`' error if `test[0]` doesn't exist. Yeah, it'd be stupid for a TestScript/TestReport not to have a single Test element, but it's technically still valid. By using the `each` method, if the number of elements in `test` is 0, it just never asks it for `action` method.

Reply:
Good point. I will commit the suggestion.